### ISSUE TITLE
Facebook download failed

### ISSUE THREAD

**Issue by Pitel** [rg3/youtube-dl#8504]
```
[debug] System config: []
[debug] User config: [u'-v', u'--console-title']
[debug] Command-line args: [u'https://www.facebook.com/jindrich.bauer.58/videos/vb.1402944783/10205466528528287/?type=2&theater']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.02.10
[debug] Python version 2.7.10 - Linux-4.2.0-23-generic-x86_64-with-Ubuntu-15.10-wily
[debug] exe versions: none
[debug] Proxy map: {}
[facebook] 10205466528528287: Downloading webpage
ERROR: Unable to download webpage: HTTP Error 500: Internal Server Error (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "/home/pitel/bin/youtube-dl/youtube_dl/extractor/common.py", line 354, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/home/pitel/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1906, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
```

Downloadingh different video from FB few minutes ago works, the video plays in browser.

**Comment by tjesp** [comment]
I got the same error with this video:

https://www.facebook.com/DavidCooper97/videos/10204648361313294/

However I was able to download another Facebook video so its not all that are dead.

**Comment by libjared** [comment]
When running OP's command, I received the same result: HTTP 500. But, **when logging in to Facebook with -u and -p flags**, that video was downloaded normally.

(On another note, that specific video description is not in my language, so the "See Translation" link was added to the description... and the filename as a result! "Máte ještě někdo neřešitelný problém See Translation-10205466528528287.mp4")

**Comment by Adastras** [comment]
Loading up the original FB url in a Chrome incognito window gives me a 500 error *most* of the time. Eventually it does load after hitting it enough times.

**Comment by yan12125** [comment]
Thanks @Adastras. For https://www.facebook.com/axel.klos.7/videos/1636612619944995/ (reported in #8578), both Firefox and Chrome on Linux leads to frequent HTTP 500 errors. I can't simply change the user-agent to a different one because Facebook serves more formats to Chrome. (See #8222 and #8374) Can anyone found an user-agent that reports DASH formats while not causing frequent HTTP 500 errors? By changing the following line in ```youtube_dl/extractor/facebook.py```:
```
_CHROME_USER_AGENT = 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.97 Safari/537.36'
```
And then run the following command:
```
$ youtube-dl -vF "https://www.facebook.com/video.php?v=957955867617029"
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-vF', 'https://www.facebook.com/video.php?v=957955867617029']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.02.13
[debug] Git HEAD: cb64ccc
[debug] Python version 3.5.1 - Linux-4.4.1-2-ARCH-x86_64-with-arch-Arch-Linux
[debug] exe versions: avconv v12_dev0-2305-g77a44f5, avprobe v12_dev0-2305-g77a44f5, ffmpeg 2.8.6, ffprobe 2.8.6, rtmpdump 2.4
[debug] Proxy map: {}
[facebook] 957955867617029: Downloading webpage
[info] Available formats for 957955867617029:
format code                      extension  resolution note
progressive_sd_src               mp4        unknown    
progressive_sd_src_no_ratelimit  mp4        unknown    
485721114944025ad                mp4        audio only DASH audio   49k , mp4a.40.29 (44100Hz)
584741191674072vd                mp4        240x426    DASH video  283k , avc1.4d401e, video only
474346612748657v                 mp4        360x640    DASH video  331k , avc1.64001f, video only (best)
```
If there are DASH audio/video formats reported, that's our goal. The format ID may be different.

**Comment by yan12125** [comment]
All videos mentioned here are working again. Seems Facebook has fixed their HTTP 500 errors. Feel free to open new issues if there are still problems.
